src/main/java/org/wildfly/security/auth/provider/SimpleMapBackedSecurityRealm.java

```diff
@@ -157,21 +159,25 @@ public AuthorizationIdentity getAuthorizationIdentity() {
             return entry == null ? AuthorizationIdentity.EMPTY : AuthorizationIdentity.basicIdentity(entry.getAttributes());
         }
 
+        @Override
         public boolean verifyCredential(final Object credential) throws RealmUnavailableException {
-            if (credential instanceof char[]) try {
-                final SimpleRealmEntry entry = map.get(name);
-                if (entry == null) {
+            try {
+                if (credential instanceof char[]) {
+                    final Password password = map.get(name).getPassword();
```

Comment:
NPE if the map doesn't contain the name; IMO the local var extraction/null check should remain.